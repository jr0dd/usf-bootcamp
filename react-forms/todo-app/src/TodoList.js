import React, { useState } from 'react'
import Todo from './Todo'
import NewTodoForm from './NewTodoForm'

function TodoList () {
  const [todos, setTodos] = useState([])

  const add = todo => {
    setTodos(todos => [...todos, todo])
  }

  const remove = id => {
    setTodos(todos => todos.filter(t => t.id !== id))
  }

  const todoMap = todos.map(t => (
    <Todo
      key={t.id}
      id={t.id}
      task={t.task}
      handleRemove={remove}
    />
  ))

  return (
    <div>
      <NewTodoForm addTodo={add} />
      <div>{todoMap}</div>
    </div>
  )
}

export default TodoList
