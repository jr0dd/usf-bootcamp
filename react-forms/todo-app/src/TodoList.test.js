import { render } from '@testing-library/react'
import TodoList from './TodoList'

it('renders without crashing', () => {
  render(<TodoList />)
})

it('matches snapshot', () => {
  const { asFragment } = render(<TodoList />)
  expect(asFragment()).toMatchSnapshot()
})
