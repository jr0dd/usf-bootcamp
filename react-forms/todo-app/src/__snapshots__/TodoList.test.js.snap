// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <form>
        <div>
          <label
            for="task"
          >
            Task
          </label>
          <input
            id="task"
            name="task"
            type="text"
            value=""
          />
        </div>
        <button
          id="add"
        >
          Add
        </button>
      </form>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
